# Week 5 Assignment

We analyze the CSV data taken from https://github.com/hadley/nycflights13/tree/master/data-raw  to answer three questions.

# 1. What is the northernmost airport in the United States?

We begin by reading in the airports.csv file from the source link.

In [1]:
import pandas as pd
url = 'https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv'
data = pd.read_csv(url)

Next we sort by latitude, and list only the top 5 results.

In [2]:
data.sort_values(by = 'lat', ascending = False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt   tz dst              tzone  
417  149 -5.0   A                 \N  
230   44 -9.0   A  America/Anchorage  
110   41 -9.0   A  America/Anchorage  
708   35 -9.0   A  America/Anchorage  
152   96 -9.0   A  America/Anchorage

It's suspicious that the top result has a longitude so much different than the other results, so we investigate further.  A Google search of 72.270833	42.898333 reveals a point in the middle of the ocean!  Another Google search of the EEN airport code shows that it refers to an airport in New Hampshire, with the proper coordinates of 42.898389 -72.270777.  Not only were the latitude and longitude transposed, but the longitude coordinate should have been negative. 
Bad data!

So we see that the northernmost airport in the United States is the next airport on the list, **Wiley Post Will Rogers (BRW).**  This aligns with our expectation that the northernmost airport would be found in Alaska.

# 2. What is the easternmost airport in the United States?

We run a similar analysis on our data, this time sorting by longitude.

In [3]:
data.sort_values(by = 'lon', ascending = False).head()

faa                           name        lat         lon   alt   tz  \
1290  SYA                   Eareckson As  52.712275  174.113620    98 -9.0   
942   MYF               Montgomery Field  32.475900  117.759000    17  8.0   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478  8.0   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149 -5.0   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45 -5.0   

     dst              tzone  
1290   A  America/Anchorage  
942    A     Asia/Chongqing  
396    A     Asia/Chongqing  
417    A                 \N  
444    A   America/New_York

This output seems strange.  The top listing is in Alaska, which is pretty far west.  The next two results show Asia/Chongqing timezone and are similiarly suspicious.  The fourth result has incomplete data.  We run the query again, but expand it to see the top 10 results for better context.

In [4]:
data.sort_values(by = 'lon', ascending = False).head(10)

faa                                 name        lat         lon   alt  \
1290  SYA                         Eareckson As  52.712275  174.113620    98   
942   MYF                     Montgomery Field  32.475900  117.759000    17   
396   DVT        Deer Valley Municipal Airport  33.411700  112.457000  1478   
417   EEN              Dillant Hopkins Airport  72.270833   42.898333   149   
444   EPM           Eastport Municipal Airport  44.910111  -67.012694    45   
624   HUL                         Houlton Intl  46.123083  -67.792056   489   
259   CAR                         Caribou Muni  46.871500  -68.017917   626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958  -68.044797   534   
1398  WFK  Northern Aroostook Regional Airport  47.285556  -68.312778   988   
192   BHB          Hancock County - Bar Harbor  44.449769  -68.361565    83   

       tz dst              tzone  
1290 -9.0   A  America/Anchorage  
942   8.0   A     Asia/Chongqing  
396   8.0   A     Asia/Chongqing  
417  -5.0   A                 \N  
444  -5.0   A   America/New_York  
624  -5.0   A   America/New_York  
259  -5.0   A   America/New_York  
1101 -5.0   A   America/New_York  
1398 -5.0   A   America/New_York  
192  -5.0   A   America/New_York

Results 5 through 10 seem to be better aligned with each other, and therefore we have more confidence to say that **Eastport Municipal Airport (EPM)** is the easternmost airport in the United States.

# 3. On February 12th, 2013, which New York area airport had the windiest weather?

We switch to the weather.csv to find our relevant data.

In [5]:
w_url = 'https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv'
columns = ['origin', 'year', 'month', 'day', 'wind_speed', 'wind_gust']
weather = pd.read_csv(w_url, usecols = columns)

Next we sort the data to show the top 5 results.

In [6]:
feb12 = ((weather['year'] == 2013) & (weather['month'] == 2) & (weather['day'] == 12))
weather[feb12].sort_values(by = 'wind_speed', ascending = False).head()

origin  year  month  day  wind_speed    wind_gust
1015     EWR  2013      2   12  1048.36058  1206.432388
18433    LGA  2013      2   12    23.01560    26.485892
1024     EWR  2013      2   12    21.86482    25.161598
18444    LGA  2013      2   12    21.86482    25.161598
9728     JFK  2013      2   12    20.71404    23.837303

This shows EWR as the top result, but with an impossible windspeed of 1048 miles per hour.  Obviously this is incorrect data, so we conclude that **LGA** is the New York area airport with the windiest weather on February 12, 2013.